src/main/scala/no/ndla/draftapi/integration/TaxonomyApiClient.scala

```diff
@@ -0,0 +1,157 @@
+      get[List[Resource]](s"$TaxonomyApiEndpoint/queries/resources", "contentURI" -> s"urn:article:$articleId")
+    }
+
+    def queryTopic(articleId: Long): Try[List[Topic]] = {
+      get[List[Topic]](s"$TaxonomyApiEndpoint/queries/topics", "contentURI" -> s"urn:article:$articleId")
+    }
+
+    def updateTaxonomyIfExists(articleId: Long, article: Article): Try[Long] = {
+      // Get places to update names
+      val resourcesAndTopics = for {
+        resources <- queryResource(articleId)
+        topics <- queryTopic(articleId)
+      } yield (resources, topics)
+
+      resourcesAndTopics match {
+        case Success((res, top)) =>
+          Language.findByLanguageOrBestEffort(article.title, Language.DefaultLanguage) match {
+            case Some(defaultTitle) =>
+              // Do the updates
+              val resources = res.map(r => updateResourceTitleAndTranslations(r, defaultTitle, article))
+              val topics = top.map(t => updateTopicTitleAndTranslations(t, defaultTitle, article))
+
+              // Log errors
+              resources
+                .collect { case Failure(ex) => ex }
+                .foreach(ex => logger.warn(s"Taxonomy update failed with: ${ex.getMessage}"))
+              topics
+                .collect { case Failure(ex) => ex }
+                .foreach(ex => logger.warn(s"Taxonomy update failed with: ${ex.getMessage}"))
+
+              // Return an error if there was one
+              val resourceResult = resources.collectFirst { case Failure(ex) => ex } match {
+                case Some(ex) => Failure(ex)
+                case None     => Success(articleId)
+              }
+
+              val topicResult = topics.collectFirst { case Failure(ex) => ex } match {
+                case Some(ex) => Failure(ex)
+                case None     => Success(articleId)
+              }
+
+              (resourceResult, topicResult) match {
+                case (Success(r), Success(t)) => Success(r)
+                case (Failure(ex), _)         => Failure(ex)
+                case (_, Failure(ex))         => Failure(ex)
+              }
+            case None => Failure(new RuntimeException("This is a bug, no name was found for published article."))
+          }
+        case Failure(ex) => Failure(ex)
+      }
+    }
+
+    private def updateResourceTitleAndTranslations(res: Resource, defaultTitle: ArticleTitle, article: Article) = {
+      val resourceResult = updateResource(res.copy(name = defaultTitle.title))
+      val translationResult = updateResourceTranslations(res.id, article.title)
+
+      (resourceResult, translationResult) match {
+        case (Success(s1), Success(_)) => Success(s1)
+        case (Failure(ex), _)          => Failure(ex)
+        case (_, Failure(ex))          => Failure(ex)
+      }
+    }
+
+    private def updateTopicTitleAndTranslations(top: Topic, defaultTitle: ArticleTitle, article: Article) = {
```

Comment:
Her ble det litt mye duplisert kode. Tipper du kan gjøre dette enklere vedå bare ha én felles funksjon for å oppdatere titler og translations. F.eks ved å sende inn funksjoner somparametre som kaller riktig endepunkter.
Gjelder både `updateResourceTitleAndTranslations`, `updateTopicTitleAndTranslations`, `updateResourceTranslations`, og `updateTopicTranslations`